### DISCUSSION TITLE
Publishing joint positions from ros to robotstudio

### DISCUSSION THREAD

**Discussion by kalepo** [ros-industrial/abb_robot_driver#63] [Q&A]
I've recently been publishing robot arm joint velocities from unity and publishing real time joint states into a rostopic.

I want to now publish those real time joint states to robotstudio to make it move in real time...

Does anyone know how I would I do this?

**Comment by gavanderhoorn** [comment]
You could use a `JointGroupPositionController` from `ros_controllers`.

You'd have to make sure your motions are continuous though, or tune the filtering EGM applies.

**Reply by kalepo** [reply]
How would I produce and send joint positions from unity to the ros_controller? I've made a urdf in unity published its joint positions to ROS

**Reply by gavanderhoorn** [reply]
That would be something for you to figure out I'm afraid.

As a general comment, `JointState` is not typically used for *commanding* robots in ROS. There are other messages to use, such as `FollowJointTrajectoryAction`.

It can be done though, but fi the `JointGroupPositionController` would require different inputs (`std_msgs::Float64MultiArray`), so you'd have to convert.

**Reply by kalepo** [reply]
How would I go about converting from sensor_msgs/JointState to std_msgs::Float64MultiArray?

**Reply by kalepo** [reply]
@gavanderhoorn any ideas for this?
